File: codeup/100000572/C.py
#!/usr/bin/python3
from typing import Union


class school_person:
    category = Union[int, str]

    def __init__(self, num, name, sex, job, category):
        self.num = num
        self.name = name
        self.sex = sex
        self.job = job
        if job == 's':
            self.category = category
        if job == 't':
            self.category = category


n = int(input())
school_person_list = []
for i in range(n):
    num, name, sex, job, category = list(input().split())
    school_person_list.append(school_person(num, name, sex, job, category))

for i in range(n):
    print(school_person_list[i].num, school_person_list[i].name,
          school_person_list[i].sex, school_person_list[i].job, school_person_list[i].category)

File: codeup/100000572/C.cpp
#include <iostream>
using namespace std;

int main() {
  struct school_person {
    int num;
    char name[10];
    char sex;
    char job;
    union {
      int my_class;
      char position[10];
    } category;
  };

  school_person list[100];
  int n;
  scanf("%d", &n);
  for (int i = 0; i < n; i++) {
    scanf("%d %s %c %c", &list[i].num, list[i].name, &list[i].sex,
          &list[i].job);
    if (list[i].job == 's')
      scanf("%d", &list[i].category.my_class);
    else if (list[i].job == 't')
      scanf("%s", list[i].category.position);
  }

  for (int i = 0; i < n; i++) {
    printf("%d %s %c %c ", list[i].num, list[i].name, list[i].sex, list[i].job);
    if (list[i].job == 's')
      printf("%d\n", list[i].category.my_class);
    else if (list[i].job == 't')
      printf("%s\n", list[i].category.position);
  }

  return 0;
}